### ISSUE TITLE
InverseKinematicsTool marker error calculation: vector size is incorrect.

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#2052]
In InverseKinematicsTool, if the number of model markers is less than the number of experimental markers, then we incorrectly read past the end of an array:

https://github.com/opensim-org/opensim-core/blob/4199ecd2f211a5063716390f73b62aff8e0cce8c/OpenSim/Tools/InverseKinematicsTool.cpp#L315-L334

Here, `nm` is the number of experimental markers, but ` ikSolver.computeCurrentSquaredMarkerErrors(squaredMarkerErrors);` resizes `squaredMarkerErrors` to be the number of model markers. The next few lines then index into `squaredMarkerErrors` as if its length were the number of experimental markers.

This could potentially cause us to report much smaller marker errors than is correct, because the RMS error is computed by dividing by the (incorrectly) larger number of experimental markers.

**Comment by aseth1** [comment]
I am not sure this assessment is correct. 
In `InverseKinematicsSolver::computeCurrentSquaredMarkerErrors()` the `squaredMarkerErrors` vector is resized here:
```cpp
markerErrors.resize(_markerAssemblyCondition->getNumMarkers());
```
and the `markerAssemblyCondition` is not the model but the internal assembly condition which only includes markers that were supplied by the `MarkersReference`.

**Comment by aseth1** [comment]
Talking with @chrisdembia the issue is that `_markerAssemblyCondition->getNumMarkers()` is the size of the intersection of experimental and model markers and `markersReference.getNumRefs()` includes all the experimental markers, regardless if they are used or not. Reporting should only be based on the intersection and not on all the markers provided.

**Comment by aseth1** [comment]
Closed by #2054
